tarogarden/caretaker.go

```diff
@@ -762,3 +769,16 @@ func GetTxFee(pkt *psbt.Packet) (int64, error) {
 
 	return inputValue - outputValue, nil
 }
+
+// GenHeaderVerifier generates a block header on-chain verification callback
+// function given a chain bridge.
+func GenHeaderVerifier(ctx context.Context,
+	chainBridge ChainBridge) func(header wire.BlockHeader) error {
+
+	return func(blockHeader wire.BlockHeader) error {
+		_, err := chainBridge.GetBlock(
+			ctx, blockHeader.BlockHash(),
+		)
+		return err
+	}
+}
```

Comment:
I'm not sure where to put this function. Any ideas?

Reply:
Maybe we can have an interface 

```
type HeaderVerifier interface {
    func GetBlock(ctx Context, hash chainhash.Hash) error
}
```

And use the `ChainBridge` as a parameter (it already implements that interface). In `func (p *Proof) Verify(...)` we call `headerVerifier(p.BlockHeader.BlockHash())`

Reply:
`GetBlock` has the signature ```GetBlock(context.Context, chainhash.Hash) (*wire.MsgBlock, error)``` in `ChainBridge`, which means that `headerVerifier` in `func (p *Proof) Verify(...)` would need to be 
```
_, err := headerVerifier(ctx, p.BlockHeader.BlockHash())
```

I'm not sure if that's better than a callback, because `headerVerifier` would then return a `*wire.MsgBlock` which seems unnecessary. It would also take a block hash rather than a block as an argument which I think makes less sense.

Am I missing something? I'm not sure if I'm looking at this in the right way.

Reply:
That rational makes sense to me. Unless we have more methods to add we can follow the function based approach :+1:

Reply:
I'm not sure if there's a reason we don't use callbacks elsewhere - I think the conversions between blockhashes and blocks could be handled by unexported methods of some type wrapping `ChainBridge`. We need to have a `ChainBridge` instance available in all places where we use `GenHeaderVerifier`, so we could also build a wrapping object, which may be easier to add to later on.